src/__init__.py

```diff
@@ -1,9 +1,17 @@
 from config import Config
 from .database.db_connection import DBConnection
+from src.anamoly_detection.anomaly_engine import AnomalyEngine
 
 run_config = Config()
 try:
     db = DBConnection(run_config.db_file)
     db.create_session()
+    anomaly_engine = AnomalyEngine(db)
 except Exception as e:
+    db = None
     run_config.log_event.info(f"Database connection error {e}")
+try:
+    anomaly_engine = AnomalyEngine(db)
+except Exception as e:
+    anomaly_engine = AnomalyEngine(None)
+    run_config.log_event.info(f"Anomaly Engine initialization error: {e}")
```

Comment:
The anomaly engine can import the db connection initialized, why does it need to be given one as a class parameter? Additionally I think this code may not work because if the original call to create_session() function works then the subsequent call may cause an error.

Reply:
Limiting a class to only function with regards to an external reference to something isn't very expandable. Allowing for the anomaly engine to specify this connection makes it more expandable, for instance if the connection would ever be moved to some online DB for equations. And the subsequent call to create_session() is only if there is already no existing session.